app/services/orphaned_profiles_cleaner.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# Service for cleaning up orphaned profiles
+class OrphanedProfilesCleaner
+  class << self
+    def run!
+      # No need to run this in an empty DB
+      return if Profile.canonical(false).none?
+
+      reassign_by_parent
+      Profile.canonical(false).where(policy_id: nil).destroy_all # Remove remaining orphaned profiles
+    end
+
+    private
+
+    def reassign_by_parent
+      ActiveRecord::Base.connection.execute(%(
+        UPDATE "profiles" SET "policy_id" = "sq"."policy_id" FROM (
+          SELECT "profiles"."id", "xp"."policy_id" FROM "profiles"
+          INNER JOIN "profiles" as "xp" ON "xp"."parent_profile_id" = "profiles"."parent_profile_id"
+                                        AND "xp"."account_id" = "profiles"."account_id"
+                                        AND "xp"."external" = false
```

Comment:
Should there be a check for `"profiles".external = true` to avoid possible two profiles under same policy being marked as the "initial policy" (with `external = false`)?

Reply:
Fixed